core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -146,6 +151,7 @@ private class ShuffleStatus(
       _numAvailableMapOutputs += 1
       invalidateSerializedMapOutputStatusCache()
     }
+    mapStatusesDeleted(mapIndex) = mapStatuses(mapIndex)
```

Comment:
This is `addMapOutput`, why we need to add it to `mapStatusesDeleted`?

Reply:
Probably also remove from `mapStatusesDeleted` if present ?

Reply:
Hi, @viirya and @mridulm .
This is based on the semantic of this function description. With this, we will recover only the last one.
```
  /**
   * Register a map output. If there is already a registered location for the map output then it
   * will be replaced by the new location.
   */
```

Reply:
(I am assuming @viirya has the same query)
Given we have a new map output, which we are storing as `mapStatuses(mapIndex) = status` in the next line, why are we saving the previous mapStatus ?
I am sure I am missing something here.

Reply:
yea, same question.

Reply:
Ah, I missed your intention. Right, AS-IS code is matching `mapStatusesDeleted` every places where `mapStatuses` is changed. I'll remove this part. Thanks, @viirya and @mridulm .